bowling/testingScenarios

```diff
@@ -0,0 +1,458 @@
+            int score = countScore(v);
+            THEN ("score is incorrect, maximum score in one throw is 10") {
+                """REQUIRE EXCEPTION"""  
+            }
+        }
+    }
+}
+
+SCENARIO("game with too many rounds"){
+    GIVEN("game with too many rounds") {
+        std::vector<int> v{0,0, 0,0, 0,0, 0,0, 0,0, 0,0, 0,0, 0,0, 0,0, 0,0, 0,0, 0,0, 0,0};
+        WHEN ("count score after game") {
+            int score = countScore(v);
+            THEN ("number of rounds is incorrect, there should be 10 rounds") {
+                """REQUIRE EXCEPTION"""  
+            }
+        }
+    }
+}
+
+SCENARIO("game with too many bonus throws"){
+    GIVEN("game with too many bonus throws") {
+        std::vector<int> v{0,0, 0,0, 0,0, 0,0, 0,0, 0,0, 0,0, 0,0, 0,0, 10,10,10,10};
+        WHEN ("count score after game") {
+            int score = countScore(v);
+            THEN ("number of bonus throws is incorrect, there should be max of two bonus throws") {
+                """REQUIRE EXCEPTION"""  
+            }
+        }
+    }
+}
+
+SCENARIO("game with too less throws in final round"){
+    GIVEN("game with too less throws in final round") {
+        std::vector<int> v{0,0, 0,0, 0,0, 0,0, 0,0, 0,0, 0,0, 0,0, 0,0, 10};
+        WHEN ("count score after game") {
+            int score = countScore(v);
+            THEN ("number of throws in final round is incorrect, there should be at least two throws in last round") {
+                """REQUIRE EXCEPTION"""
+            }
+        }
+    }
+}
+
+SCENARIO("game with too less rounds"){
+    GIVEN("game with too less rounds") {
+        std::vector<int> v{0,0, 0,0};
+        WHEN ("count score after game") {
+            int score = countScore(v);
+            THEN ("number of rounds is too small") {
+                """REQUIRE EXCEPTION"""
+            }
+        }
+    }
+}
+
+SCENARIO("check score vector size with no spare or strike at the end"){
+    GIVEN("sccore vector of pairs") {
+        std::vector<std::pair<int,int>> v {};
+        const size_t rounds = 10;
+        WHEN ("check vector size after game") {
+            playGame(rounds);
+            THEN ("vector size should be equal 10 or 11 or 12") {
+                if (v.back().first == 10) { 
+                    REQUIRE_THAT(v.size(), Equals(12));
+                }
+                else if (v.back().first + v.back().second == 10) {
+                    REQUIRE_THAT(v.size(), Equals(11));
+                }
+                REQUIRE_THAT(v.size(), Equals(10));
+            }
+        }
+    }
+}
+
+SCENARIO("check all vector elements are digit"){
+    GIVEN("empty vector of pairs") {
+        std::vector<pair<int,int>> v{};
+        WHEN ("check all elements after game is played") {
+            const size_t rounds = 10;
+            playGame(rounds);
+            THEN ("all elements of score vector should be digit") {
+                auto& it = find_if_not(v.begin(), v.end(), isdigit);
```

Comment:
type int doesn't allow anything other that numbers to be stored. std::isdigit works on char.